File: NendVideoMediationAdapterSample/NendVideoMediationAdapterSample/ViewController.m
//
//  ViewController.m
//  NendVideoMediationAdapterSample
//
//  Copyright © 2017年 F@N Communications, Inc. All rights reserved.
//

#import "ViewController.h"
@import GoogleMobileAds;
@import NendMediationCustomEventAdapter;

@interface ViewController () <GADRewardBasedVideoAdDelegate>

@property (nonatomic) NSString *unitId;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.unitId = @"your unitId";
    
    [GADRewardBasedVideoAd sharedInstance].delegate = self;
}

- (IBAction)loadAd:(id)sender {
    GADRequest *request = [GADRequest request];
    GADMAdapterNendCustomEventExtras *extras = [[GADMAdapterNendCustomEventExtras alloc] init];
    extras.userId = @"ID used in games and applications";
    [request registerAdNetworkExtras:extras];
    [[GADRewardBasedVideoAd sharedInstance] loadRequest:request withAdUnitID:self.unitId];
}

- (IBAction)showAd:(id)sender {
    if ([[GADRewardBasedVideoAd sharedInstance] isReady]) {
        [[GADRewardBasedVideoAd sharedInstance] presentFromRootViewController:self];
    }
}

#pragma mark GADRewardBasedVideoAdDelegate
/// Tells the delegate that the reward based video ad has rewarded the user.
- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didRewardUserWithReward:(GADAdReward *)reward {
    NSLog(@"%s %@: %d", __FUNCTION__, reward.type, reward.amount.intValue);
}

/// Tells the delegate that the reward based video ad failed to load.
- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didFailToLoadWithError:(NSError *)error {
    NSLog(@"%s error: %@", __FUNCTION__, error);
}

/// Tells the delegate that a reward based video ad was received.
- (void)rewardBasedVideoAdDidReceiveAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"%s", __FUNCTION__);
}

/// Tells the delegate that the reward based video ad opened.
- (void)rewardBasedVideoAdDidOpen:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"%s", __FUNCTION__);
}

/// Tells the delegate that the reward based video ad started playing.
- (void)rewardBasedVideoAdDidStartPlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"%s", __FUNCTION__);
}

/// Tells the delegate that the reward based video ad closed.
- (void)rewardBasedVideoAdDidClose:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"%s", __FUNCTION__);
}

/// Tells the delegate that the reward based video ad will leave the application.
- (void)rewardBasedVideoAdWillLeaveApplication:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"%s", __FUNCTION__);
}

@end

File: NendVideoMediationAdapterSample/NendVideoMediationAdapterSample_Swift/ViewController.swift
//
//  ViewController.swift
//  NendVideoMediationAdapterSample_Swift
//
//  Copyright © 2017年 F@N Communications, Inc. All rights reserved.
//

import UIKit
import GoogleMobileAds
import NendMediationCustomEventAdapter

class ViewController: UIViewController {
    
    let unitId = "your unitId"

    override func viewDidLoad() {
        super.viewDidLoad()
        
        GADRewardBasedVideoAd.sharedInstance().delegate = self
    }

    @IBAction func loadAd(_ sender: Any) {
        let request = GADRequest()
        let extras = GADMAdapterNendExtras()
        extras.userId = "ID used in games and applications"
        request.register(extras)
        GADRewardBasedVideoAd.sharedInstance().load(request, withAdUnitID: self.unitId)
    }

    @IBAction func showAd(_ sender: Any) {
        if (GADRewardBasedVideoAd.sharedInstance().isReady) {
            GADRewardBasedVideoAd.sharedInstance().present(fromRootViewController: self)
        }
    }
}

extension ViewController : GADRewardBasedVideoAdDelegate {
    // MARK: - GADRewardBasedVideoAdDelegate
    /// Tells the delegate that the reward based video ad has rewarded the user.
    public func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd, didRewardUserWith reward: GADAdReward) {
        print("\(#function) \(reward.type) : \(reward.amount)")
    }
    
    /// Tells the delegate that the reward based video ad failed to load.
    public func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd, didFailToLoadWithError error: Error) {
        print("\(#function) error: \(error)")
    }
    
    /// Tells the delegate that a reward based video ad was received.
    public func rewardBasedVideoAdDidReceive(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print(#function)
    }
    
    /// Tells the delegate that the reward based video ad opened.
    public func rewardBasedVideoAdDidOpen(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print(#function)
    }
    
    /// Tells the delegate that the reward based video ad started playing.
    public func rewardBasedVideoAdDidStartPlaying(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print(#function)
    }
    
    /// Tells the delegate that the reward based video ad closed.
    public func rewardBasedVideoAdDidClose(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print(#function)
    }
    
    /// Tells the delegate that the reward based video ad will leave the application.
    public func rewardBasedVideoAdWillLeaveApplication(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print(#function)
    }
}

